Add alert when wc-admin build does not exist.

timmyc opened the pull request:
Fixes #424 

Currently if the repository is cloned to the plugins directory of a WordPress install, and activated without being built, a variety of errors are shown on the screen. This branch borrows logic from the WooCommerce Blocks repository to display an admin_notice when this happens.

### Screenshots

![dashboard woo test wordpress 2018-12-12 09-53-18](https://user-images.githubusercontent.com/22080/49888860-e6c02600-fdf4-11e8-81c8-ccc5672ac210.png)

Updated after feedback:

![image](https://user-images.githubusercontent.com/22080/49956709-c195eb00-feba-11e8-981e-b2f29b09308f.png)

### Detailed test instructions:

- Simulate wc-admin not being built by either `rm`ing your local build directory or doing a fresh clone with `master`
- Activate the plugin, and verify the files not existing errors are shown
- Apply this patch, verify the admin notice is shown
- run `npm start` and verify wc-admin loads as expected

timmyc committed:
Add alert when wc-admin build does not exist.

timmyc commented:
Question: Should we remove the `WP_DEBUG` bit and run this logic regardless? Guessing some users might not have that constant set locally.

justinshreve commented on wc-admin.php:
```diff
@@ -49,6 +49,15 @@ function wc_admin_plugins_notice() {
+ */
+function wc_admin_build_notice() {
+	echo '<div class="error"><p>';
+	esc_html_e( 'WooCommerce Admin development mode requires files to be built. From the plugin directory, run <code>npm install</code> to install dependencies, <code>npm run build</code> to build the files or <code>npm start</code> to build the files and watch for changes.', 'wc-admin' );
```
You are escaping all HTML here, so `<code>` doesn't end up working:

<img width="915" alt="screen shot 2018-12-12 at 3 16 26 pm" src="https://user-images.githubusercontent.com/689165/49896286-eccbfc00-fe20-11e8-99bc-4c082bbb597b.png">

timmyc replied:
Ha, thanks for pointing that out 🤠

emdashcodes reviewed (commented):
I see the notice after renaming my dist folder. Just running into one issue.

> Question: Should we remove the WP_DEBUG bit and run this logic regardless? Guessing some users might not have that constant set locally.

Yes, I think we should. The users who try to pull this down without building/looking at the README will probably also not have the correct things flagged.

joshuatf reviewed (commented):
Testing well for me as well! 👍 

>Should we remove the WP_DEBUG bit and run this logic regardless? Guessing some users might not have that constant set locally.

Definitely.  We might even consider changing the text to something along the lines of `You've downloaded the WooCommerce Admin development version...`.  My fear is that not-so-super-tech-savvy people come along and grab this repo zip from somewhere without `WP_DEBUG` enabled and frustration ensues.

timmyc committed:
Updates per feedback.

timmyc commented:
Thanks @joshuatf and @justinshreve for the feedback. I applied some changes to make the message appear regardless of WP_DEBUG status, and changed the verbiage a bit to make it more non-tech savvy and included a link to the releases page so folks can get a pre-built version.

I also moved the logic that displays the message and `return`s to ensure the test suite will still have the requisite classes to load and run properly.

joshuatf commented on wc-admin.php:
```diff
@@ -49,6 +49,19 @@ function wc_admin_plugins_notice() {
+	$message_one = __( 'You have installed a development version WooCommerce Admin which requires files to be built. From the plugin directory, run <code>npm install</code> to install dependencies, <code>npm run build</code> to build the files.', 'wc-admin' );
+	$message_two = sprintf(
+		/* translators: 1: URL of GitHub Repository build page */
+		__( 'Or you can download a pre-built version of the plugin by visiting <a href="%1$s">the releases page in the repository</a>.', 'wc-admin' ),
```
Awesome ❤️

joshuatf commented on wc-admin.php:
```diff
@@ -49,6 +49,19 @@ function wc_admin_plugins_notice() {
+ * Notify users that the plugin needs to be built
+ */
+function wc_admin_build_notice() {
+	$message_one = __( 'You have installed a development version WooCommerce Admin which requires files to be built. From the plugin directory, run <code>npm install</code> to install dependencies, <code>npm run build</code> to build the files.', 'wc-admin' );
```
Nice!  One small fix here:

>You have installed a development version **of** WooCommerce Admin

joshuatf reviewed (approved):
Looks good and still tests well!  One tiny `of` addition, but pre-approving ✅

timmyc committed:
Typo and fix for ci ( i hope )
